language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeValidator.java

```diff
@@ -0,0 +1,304 @@
+        }
+        visited.add(node);
+        
+        return !node.equalsToken().get().isMissing()
+                && node.typedBindingPattern().apply(this)
+                && node.initializer().isPresent()
+                && node.initializer().get().apply(this);
+    }
+
+    @Override
+    public Boolean transform(FunctionCallExpressionNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        if (node.functionName().isMissing() || node.openParenToken().isMissing()
+                || node.closeParenToken().isMissing()
+                || !node.arguments().stream().allMatch(arg -> arg.apply(this))) {
+            return false;
+        }
+        // Check for missing commas
+        for (int i = 0; i < node.arguments().separatorSize(); i++) {
+            if (node.arguments().getSeparator(i).isMissing()) {
+                return false;
+            }
+        }
+        return node.parent().apply(this);
+    }
+    
+    @Override
+    public Boolean transform(TypedBindingPatternNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        return node.bindingPattern().apply(this)
+                && node.typeDescriptor().apply(this);
+    }
+
+    @Override
+    public Boolean transform(CaptureBindingPatternNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        return !node.variableName().isMissing();
+    }
+    
+    @Override
+    public Boolean transform(CheckExpressionNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        return !node.checkKeyword().isMissing()
+            && node.expression().apply(this);
+    }
+    
+    @Override
+    public Boolean transform(TableTypeDescriptorNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        return !node.tableKeywordToken().isMissing()
+                && node.rowTypeParameterNode().apply(this)
+                && node.keyConstraintNode().isPresent() 
+                && node.keyConstraintNode().get().apply(this);
+    }
+    
+    @Override
+    public Boolean transform(TypeParameterNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        return !node.ltToken().isMissing()
+            && !node.gtToken().isMissing()
+            && node.typeNode().apply(this);
+    }
+
+    @Override
+    public Boolean transform(KeySpecifierNode node) {
+        if (visited.contains(node)) {
+            return true;
+        }
+        visited.add(node);
+        return !node.keyKeyword().isMissing() 
+                && !node.openParenToken().isMissing()
+                && !node.closeParenToken().isMissing()
+                && node.fieldNames().stream().noneMatch(arg->arg.isMissing());
```

Comment:
```suggestion
                && node.fieldNames().stream().noneMatch(Node::isMissing());
```
Shall we replace lambda with method reference?